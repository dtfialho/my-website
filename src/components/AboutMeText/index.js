import React from 'react'
import { useStaticQuery, graphql } from 'gatsby'
import * as S from './styled'

const AboutMeText = () => {
  const { me } = useStaticQuery(
    graphql`
      query {
        me: file(relativePath: { eq: "me.jpeg" }) {
          childImageSharp {
            fixed(width: 200, height: 200) {
              ...GatsbyImageSharpFixed
            }
          }
        }
      }
    `
  )

  return (
    <S.AboutMeTextContainer>
      <S.AboutMeFigure>
        <S.AboutMeImage fixed={me.childImageSharp.fixed} />
      </S.AboutMeFigure>
      
      <S.AboutMeTextParagraph>
        Olá! Meu nome é Diego Teixeira Fialho, nasci em Petrópolis/RJ e atualmente trabalho como Frontend Developer na <a href="http://www.personare.com.br" target="_blank" rel="noopener noreferrer">Personare</a>. Já trabalhei como Fullstack Developer na <a href="https://www.maxisite.net/" target="_blank" rel="noopener noreferrer">Maxisite</a> e Frontend Developer na <a href="https://www.alterdata.com.br/" target="_blank" rel="noopener noreferrer">Alterdata Software</a>.
      </S.AboutMeTextParagraph>

      <S.AboutMeTextParagraph>
        Apaixonado por tecnologia desde cedo sempre fui curioso para aprender. Aos 20 anos, fiz meu primeiro curso na área de tecnologia no SENAI e aos 23 ingressei na faculdade de Tecnologia da Informação e Comunicação pela FAETERJ - Petrópolis.
      </S.AboutMeTextParagraph>

      <S.AboutMeTextParagraph>
        Amo desenvolver e estudar novas tecnologias que surgem a cada dia para facilitar a nossa vida como desenvolvedores. Com o sangue metade café e metade código (e um pouco de World of Warcraft que é o meu passatempo favorito btw), amo criar e idealizar componentes utilizando frameworks Javascript e conceitos para um código limpo e bem estruturado.
      </S.AboutMeTextParagraph>
    </S.AboutMeTextContainer>
  )
}

export default AboutMeText
