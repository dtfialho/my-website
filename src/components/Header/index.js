import React from 'react'
import PropTypes from 'prop-types'
import Logo from '../Logo'
import Navbar from '../Navbar'
import * as S from './styled'

const Header = ({ isFixed }) => (
  <S.HeaderElement isFixed={isFixed}>
    <S.HeaderLogoLink to="/">
      <Logo />
    </S.HeaderLogoLink>
    <Navbar />
  </S.HeaderElement>
)

Header.propTypes = {
  isFixed: PropTypes.bool
}

export default Header
