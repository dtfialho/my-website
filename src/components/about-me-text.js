import React from 'react'
import me from '../images/me.jpeg'

const AboutMeText = () => (
  <div className="about-me__text">
    <figure className="me">
      <img src={me} alt="Diego T. Fialho" />
    </figure>
    
    <p>
      Olá! Meu nome é Diego Teixeira Fialho, nasci em Petrópolis/RJ e atualmente trabalho como Frontend Developer na <a href="http://www.personare.com.br" target="_blank" rel="noopener noreferrer">Personare</a>. Já trabalhei como Fullstack Developer na <a href="https://www.maxisite.net/" target="_blank" rel="noopener noreferrer">Maxisite</a> e Frontend Developer na <a href="https://www.alterdata.com.br/" target="_blank" rel="noopener noreferrer">Alterdata Software</a>.
    </p>

    <p>
      Apaixonado por tecnologia desde cedo sempre fui curioso para aprender. Aos 20 anos, fiz meu primeiro curso na área de tecnologia no SENAI e aos 23 ingressei na faculdade de Tecnologia da Informação e Comunicação pela FAETERJ - Petrópolis.
    </p>

    <p>
      Amo desenvolver e estudar novas tecnologias que surgem a cada dia para facilitar a nossa vida como desenvolvedores. Com o sangue metade café e metade código (e um pouco de World of Warcraft que é o meu passatempo favorito btw), amo criar e idealizar componentes utilizando frameworks Javascript e conceitos para um código limpo e bem estruturado.
    </p>
  </div>
)

export default AboutMeText
