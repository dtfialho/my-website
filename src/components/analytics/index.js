import Script from 'next/script'
import { Analytics as VercelAnalytics } from '@vercel/analytics/react'

const Analytics = () => {
  const GA_TRACKING_ID = process.env.NEXT_PUBLIC_GA_TRACKING

  return (
    <>
      <VercelAnalytics />
      <Script
        strategy="afterInteractive"
        src={`https://www.googletagmanager.com/gtag/js?id=${GA_TRACKING_ID}`}
      />
      <Script
        id="google-analytics"
        strategy="afterInteractive"
        dangerouslySetInnerHTML={{
          __html: `
            window.dataLayer = window.dataLayer || [];
            function gtag(){dataLayer.push(arguments);}
            gtag('js', new Date());
        
            gtag('config', '${GA_TRACKING_ID}');
          `
        }}
      />
    </>
  )
}

export default Analytics
