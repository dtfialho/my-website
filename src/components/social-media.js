import React from "react"
import { StaticQuery, graphql } from "gatsby"
import {
  SocialMediaContainer,
  SocialMediaItem, 
  SocialMediaLink
} from "../styles/social-media"

const SocialMedia = () => (
  <StaticQuery
    query={graphql`
      query SocialMedia {
        allSocialMediaJson {
          nodes {
            label
            icon
            link
            id
          }
        }
      }
    `}
    render={({ allSocialMediaJson: { nodes } }) => (
      <SocialMediaContainer>
        {
          nodes.map(({ link, icon, id, label }) => (
            <SocialMediaItem key={id}>
              <SocialMediaLink href={link} target="_blank" rel="noopener noreferrer" title={label}>
                <i className={icon} />
              </SocialMediaLink>
            </SocialMediaItem>
          ))
        }
      </SocialMediaContainer>
      )
    }
  />
)

export default SocialMedia
