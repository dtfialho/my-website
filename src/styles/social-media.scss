@import './utils';

.social-media {
  padding: 0;
  margin: 0;
  list-style-type: none;

  &__items {
    display: inline-block;
    margin: 8px;
  }

  &__links {
    font-size: 2rem;
    display: flex;
    width: 70px;
    height: 70px;
    color: $textColorLight;
    border-radius: 50%;
    border: 1px solid $textColorLight;
    align-items: center;
    justify-content: center;
    text-decoration: none;
    transition: .3s ease-in-out;
    backface-visibility: hidden;

    &:hover {
      background: rgba($textColorLight, .3);
    }
  }
}
