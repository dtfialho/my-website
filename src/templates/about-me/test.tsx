import { screen, render } from '@testing-library/react'
import * as Header from 'components/header'
import * as SocialMedia from 'components/social-media'
import Template from './'

jest.mock('components/header')
const mockedHeader = Header.default as jest.Mock
jest.mock('components/social-media')
const mockedContact = SocialMedia.default as jest.Mock

describe('Templates/AboutMe', () => {
  it('Should render correctly', () => {
    mockedHeader.mockImplementation(() => <header>Header</header>)
    mockedContact.mockImplementation(() => <section>Social media</section>)

    const { container } = render(<Template />)
    expect(
      screen.getByRole('img', { name: /Minha foto de perfil/ })
    ).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })
})
