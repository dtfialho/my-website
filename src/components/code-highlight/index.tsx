import { LightAsync as SyntaxHighlighter } from 'react-syntax-highlighter'
import { dracula } from 'react-syntax-highlighter/dist/cjs/styles/hljs'

type CodeHighlightProps = {
  language?: string
  value: string
}

const CodeHighlight = ({ language = '', value }: CodeHighlightProps) => (
  <SyntaxHighlighter
    language={language}
    style={dracula}
    customStyle={{ paddingLeft: '16px', paddingRight: '16px' }}
  >
    {value}
  </SyntaxHighlighter>
)

export default CodeHighlight
