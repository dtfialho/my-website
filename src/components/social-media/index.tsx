import { Github } from '@styled-icons/fa-brands/Github'
import { Linkedin } from '@styled-icons/fa-brands/Linkedin'
import { Twitter } from '@styled-icons/fa-brands/Twitter'
import * as S from './styles'

const SocialMedia = () => (
  <S.Container>
    <S.Item>
      <S.Link
        href="https://github.com/dtfialho"
        target="_blank"
        rel="noopener noreferrer me"
        title="Github"
      >
        <Github title="Ícone do Github" size={30} />
      </S.Link>
    </S.Item>

    <S.Item>
      <S.Link
        href="https://www.linkedin.com/in/diego-teixeira-fialho-35b58ab0/"
        target="_blank"
        rel="noopener noreferrer me"
        title="Linkedin"
      >
        <Linkedin title="Ícone do Linkedin" size={30} />
      </S.Link>
    </S.Item>

    <S.Item>
      <S.Link
        href="https://twitter.com/dtfialho"
        target="_blank"
        rel="noopener noreferrer me"
        title="Twitter"
      >
        <Twitter title="Ícone do Twitter" size={30} />
      </S.Link>
    </S.Item>
  </S.Container>
)

export default SocialMedia
