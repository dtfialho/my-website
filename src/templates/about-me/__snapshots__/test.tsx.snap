// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Templates/AboutMe Should render correctly 1`] = `
.c0 {
  max-width: 1200px;
  margin: 0 auto;
  padding: 153px 16px 100px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-bottom: 20px;
}

.c1 > span {
  width: 200px;
  height: 200px;
  border-radius: 50%;
}

.c2 {
  text-align: center;
  margin: 20px 0 0;
}

.c3 {
  margin: 50px 0 40px;
  border-bottom: 1px solid rgb(153,46,36);
}

.c5 {
  margin-bottom: 40px;
}

.c6 {
  margin-bottom: 10px;
  padding-bottom: 5px;
  border-bottom: 1px solid rgb(153,46,36);
}

.c7 {
  list-style-type: none;
}

.c8 {
  text-align: center;
}

.c9 {
  color: #333;
  text-align: center;
}

@media (min-width:768px) {
  .c0 {
    padding: 193px 16px 100px;
  }
}

@media (min-width:768px) {
  .c1 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    margin-bottom: 40px;
  }
}

@media (min-width:768px) {
  .c2 {
    margin-top: 0;
    margin-left: 50px;
  }
}

@media (min-width:768px) {
  .c4 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    width: 100%;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
    -webkit-flex-wrap: wrap;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
  }
}

@media (min-width:768px) {
  .c5 {
    -webkit-flex-basis: 30%;
    -ms-flex-preferred-size: 30%;
    flex-basis: 30%;
  }
}

<div>
  <header>
    Header
  </header>
  <main
    class="c0"
  >
    <section
      class="c1"
    >
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; width: 200px; height: 200px; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
      >
        <img
          alt="Minha foto de perfil"
          data-nimg="fixed"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
      <h1
        class="c2"
      >
        Sobre mim
      </h1>
    </section>
    <p>
      Olá! Meu nome é Diego Teixeira Fialho, nasci em Petrópolis/RJ e atualmente trabalho como Frontend Developer na
       
      <a
        href="http://www.personare.com.br"
        rel="noopener noreferrer"
        target="_blank"
      >
        Personare
      </a>
      . Já trabalhei como Fullstack Developer na
       
      <a
        href="https://www.maxisite.net/"
        rel="noopener noreferrer"
        target="_blank"
      >
        Maxisite
      </a>
       
      e Frontend Developer na
       
      <a
        href="https://www.alterdata.com.br/"
        rel="noopener noreferrer"
        target="_blank"
      >
        Alterdata Software
      </a>
      .
    </p>
    <p>
      Apaixonado por tecnologia desde cedo sempre fui curioso para aprender. Aos 20 anos, fiz meu primeiro curso na área de tecnologia no SENAI e aos 23 ingressei na faculdade de Tecnologia da Informação e Comunicação pela FAETERJ - Petrópolis.
    </p>
    <p>
      Amo desenvolver e estudar novas tecnologias que surgem a cada dia para facilitar a nossa vida como desenvolvedores. Com o sangue metade café e metade código, amo criar e idealizar componentes utilizando frameworks Javascript e conceitos para um código limpo e bem estruturado.
    </p>
    <div
      class="c3"
    >
      <h2>
        Um pouco do que eu faço:
      </h2>
      <div
        class="c4"
      >
        <div
          class="c5"
        >
          <h3
            class="c6"
          >
            Frontend
          </h3>
          <ul
            class="c7"
          >
            <li>
              Angular
            </li>
            <li>
              AngularJS
            </li>
            <li>
              CSS3
            </li>
            <li>
              HTML5
            </li>
            <li>
              Javascript
            </li>
            <li>
              Pré-processadores (sass, less, stylus)
            </li>
            <li>
              React
            </li>
            <li>
              Styled Components
            </li>
            <li>
              Next JS
            </li>
          </ul>
        </div>
        <div
          class="c5"
        >
          <h3
            class="c6"
          >
            Backend
          </h3>
          <ul
            class="c7"
          >
            <li>
              Node
            </li>
            <li>
              PHP
            </li>
            <li>
              Programação Orientada a Objetos
            </li>
            <li>
              Python
            </li>
            <li>
              MySql
            </li>
            <li>
              PostgreeSQL
            </li>
          </ul>
        </div>
        <div
          class="c5"
        >
          <h3
            class="c6"
          >
            Outros
          </h3>
          <ul
            class="c7"
          >
            <li>
              Scrum
            </li>
            <li>
              Sistemas de controle de versão (git)
            </li>
            <li>
              TDD
            </li>
            <li>
              Wordpress
            </li>
          </ul>
        </div>
      </div>
    </div>
    <h2
      class="c8"
    >
      Contato
    </h2>
    <div
      class="c9"
    >
      <section>
        Social media
      </section>
    </div>
  </main>
</div>
`;
